src/components/app-icon-button/AppIconButton.tsx

```diff
@@ -0,0 +1,21 @@
+import { IconButton, IconButtonProps } from '@mui/material'
+import { FC } from 'react'
+import { Link, To } from 'react-router-dom'
+
+interface AppIconButtonProps extends IconButtonProps {
+  to?: To
+}
+
+const AppIconButton: FC<AppIconButtonProps> = ({ children, to, ...props }) => {
+  if (to) {
+    return (
+      <IconButton component={Link} to={to} {...props}>
+        {children}
+      </IconButton>
+    )
+  }
+
+  return <IconButton {...props}>{children}</IconButton>
```

Comment:
You can omit children in props destructuring and refactor here like following:

```suggestion
  return <IconButton {...props} />
```

Reply:
Refactored, thanks for suggestion.

Reply:
Now it became better!